packages/gatsby/src/redux/actions/public.js

```diff
@@ -879,10 +877,18 @@ const createNode = (
     }
   }
 
+  const setTypeOwnerAction = {
+    type: `SET_TYPE_OWNER`,
+    plugin,
+    payload: {
+      typeName: node.internal.type,
+    },
+  }
```

Comment:
adding explicit action for this makes it easy to forget dispatching it everywhere it should and it might get out of sync over time 

something to consider here is that instead of new action, `pluginNamesToOwnedNodeTypes` reducer could handle `CREATE_NODE` and `DELETE_NODE` actions as they should have all the needed information attached?

Reply:
ok, yeah that makes a lot more sense. I'll make that change, thank you!